virttest/qemu_vm.py

```diff
@@ -548,17 +548,26 @@ def process_sandbox(devices, action):
                 return sandbox_option
             if action == "add":
                 sandbox_option = " -sandbox on"
+                _params = {
+                    "elevateprivileges": params.get(
+                        "qemu_sandbox_elevateprivileges", "deny"
+                    ),
+                    "obsolete": params.get("qemu_sandbox_obsolete", "deny"),
+                    "resourcecontrol": params.get(
+                        "qemu_sandbox_resourcecontrol", "deny"
+                    ),
+                    "spawn": params.get("qemu_sandbox_spawn", "deny"),
+                }
             elif action == "rem":
                 sandbox_option = " -sandbox off"
-            _params = {
-                "elevateprivileges": params.get("qemu_sandbox_elevateprivileges"),
-                "obsolete": params.get("qemu_sandbox_obsolete"),
-                "resourcecontrol": params.get("qemu_sandbox_resourcecontrol"),
-                "spawn": params.get("qemu_sandbox_spawn"),
-            }
+                _params = {
+                    "elevateprivileges": params.get("qemu_sandbox_elevateprivileges"),
+                    "obsolete": params.get("qemu_sandbox_obsolete"),
+                    "resourcecontrol": params.get("qemu_sandbox_resourcecontrol"),
+                    "spawn": params.get("qemu_sandbox_spawn"),
+                }
             for opt, val in _params.items():
-                if val is not None:
-                    sandbox_option += f",{opt}={val}"
+                sandbox_option += f",{opt}={val}"
```

Comment:
we need to check val is None or not, for when sandbox is disabled, the val can be None if we don't set it

Reply:
Yes, you're right. This part has been restored. Thank you!